### PR TITLE
Upgrade mediawiki-codesniffer to latest version (45.0.0)

### PR DESCRIPTION
**Pull request by codders**
This repository is currently using version 34 of the Mediawiki coding style phpcs rules. It should be upgraded to the latest version.

Besides being simply different, the newer versions of the rules introduce, for example,
MediaWiki.Usage.NullableType.ExplicitNullableTypes, which enforces compliance with coming deprecation rules in PHP 8.4 concerning nullable types. If the code does not remove the soon-to-be-deprecated form of nullable type declarations, the library will no longer be usable in Mediawiki for PHP 8.4 deployments.

Bug: T379481

### CODE REVIEW

**Review by lucaswerkmeister — APPROVED** [review]
LGTM, though I don’t know if we want to merge this now or wait for the CI story to clear up.

**Review by lucaswerkmeister — APPROVED** [review]

**Review by manicki — APPROVED** [review]
looks great.
Testing story is hopefully resolved through https://github.com/wmde/Time/pull/169
Can have a look at the test run for this PR dumbly merged on the CI PR: https://github.com/wmde/Time/actions/runs/12265937550/job/34223058506 (tests pass, no PHP 8.4 warnings)

**Pull request merged**
